calyx/src/passes/well_formed.rs

```diff
@@ -208,6 +252,42 @@ impl Visitor for WellFormed {
                 }
             })?;
 
+        if let CellType::Component { name: id } = &cell.prototype {
+            let mut cellmap:LinkedHashMap<ir::Id, CellType> = LinkedHashMap::new();
+            for comp in comps.iter(){
+                if comp.name.eq(id) {
+                    for cell in comp.cells.iter() {
+                        if cell.borrow().is_external() {
+                            cellmap.insert(cell.clone_name(), cell.borrow().prototype.clone());
+                        }
+                    }
+                    break;
```

Comment:
This is materializing the signature of comp every time an invoke is checked which is going to be expensive. Instead, we should pre compute all the signatures using the ConstructVisitor trait and save it into the pass struct and reuse the information when needed.